### PR TITLE
📝 Update comments in bedleveling/probe - feature/gcode 

### PR DESCRIPTION
**Pull request by classicrocker883**
### Description
Most changes are to **feature/bedlevel/**, **gcode/bedlevel/**, and **gcode/probe/**

Major changes to the comment blocks having details about Gcode commands. Updated with doxygen commands with better understand and readability.

Updates and changes including but not limited to:
- Spacing
- Remove periods at end and capitialize first word in sentence (like rest of code)
- Update/add doxygen comment commands
- Update/improve Gcode descriptions. Better readability, ect...

A couple files had some outliers that needed fixed, like `**/` instead of `*/`

### Benefits
Improves readability, Updates descriptions/details/usage of the gcodes while adding doxygen commands.

### Related Issues
in **feature/bedlevel/ubl/ubl_G29.cpp** (and also in MarlinDocumentation) one thing stood out that is wrong
```
G29 Z C uses the Current location (instead of bed center or nearest edge).
```
UBL G29 doesnt have anything that reads for 'Z' so not sure what this is about.

### CODE REVIEW

File: Marlin/src/feature/bedlevel/mbl/mesh_bed_leveling.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@
   mesh_bed_leveling::mesh_bed_leveling() {
     for (uint8_t i = 0; i < GRID_MAX_POINTS_X; ++i)
       index_to_xpos[i] = MESH_MIN_X + i * (MESH_X_DIST);
-    for (uint8_t i = 0; i < GRID_MAX_POINTS_Y; ++i)
-      index_to_ypos[i] = MESH_MIN_Y + i * (MESH_Y_DIST);
+    for (uint8_t j = 0; j < GRID_MAX_POINTS_Y; ++j)
+      index_to_ypos[j] = MESH_MIN_Y + j * (MESH_Y_DIST);
```
Comment (sjasonsmith) [inline Marlin/src/feature/bedlevel/mbl/mesh_bed_leveling.cpp]: This change doesn't belong in this PR. I'd argue that it isn't even correct, since these are not nested loops.

```suggestion
    for (uint8_t i = 0; i < GRID_MAX_POINTS_Y; ++i)
      index_to_ypos[i] = MESH_MIN_Y + i * (MESH_Y_DIST);
```

**Comment by sjasonsmith** [comment]
I reverted the code change that was mixed into this. The rest will have to wait for @thinkyhead to make the call on whether he wants these comment changes or not.

File: Marlin/src/feature/bedlevel/mbl/mesh_bed_leveling.cpp (hunk 1 above)
Comment (classicrocker883) [reply Marlin/src/feature/bedlevel/mbl/mesh_bed_leveling.cpp]: ok that makes sense, but having that most of the other for loops specifically use "i" for "x" and "j" for "y", and to argue that it *is* correct, and consistent with that logic, take a look at this for example:
in file **Marlin\src\lcd\e3v2\marlinui\ui_common.cpp**
```y
      for (uint8_t j = 0; j < (GRID_MAX_POINTS_Y); j++, y += pixels_per_y_mesh_pnt)
        for (uint8_t i = 0, x = sx; i < (GRID_MAX_POINTS_X); i++, x += pixels_per_x_mesh_pnt)
```
albeit this ex. is nested, however it uses "j" first for "y". so it makes sense to remain relative otherwise it can be very easy to be confused as to which is which.

what about in **Marlin\src\feature\encoder_i2c.cpp** 
`for (uint8_t j = 31; j > 0; j--)`

it is not nested and the others in the same file use `i`. that must be a typo then.

and in **M48.cpp** line 244, this also uses `j` for some reason.

---
<br>

off topic, but i'm unable to find a sufficient answer to this line of code in **powerless.cpp**, maybe you can help?
```y
PROCESS_SUBCOMMANDS_NOW(TS(F("M420S"), '0' + (char)info.flag.leveling, 'Z', p_float_t(info.fade, 1)));
```
this part in particular:
`TS(F("M420S"), '0' + (char)info.flag.leveling, 'Z',` 
wouldn't this result in it being "M420S0#Z...." as in (#) either '1' or '0' - the bool flag (to restore leveling)

instead would it be better to make it
`(char)(0 + info.flag.leveling), F("Z"),`

I also included the flashstringhelper for 'Z' or is this only necessary in front of actual Gcode commands?
that is another part of my question.
as you can see in this line
`PROCESS_SUBCOMMANDS_NOW(TS('T', e, 'S'))`

so like, if i were to have it like this i get a warning:
`gcode.process_subcommands_now(TS('G28XYO'));`
" warning: character constant too long for its type"

**Comment by thinkyhead** [comment]
The use of @brief, @description, @param, etc. in the descriptions of G-code parameters is improper. Those changes will all need to be reverted.
